src/Driver/ChromeDriver/Downloader.php

```diff
@@ -170,12 +171,19 @@ private function extractArchive(string $archive): string
         $unzipLocation  = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'chromedriver';
         $extractedFiles = $this->archiveExtractor->extract($archive, $unzipLocation);
 
-        $count = count($extractedFiles);
-        if ($count !== 1) {
-            throw new UnexpectedValueException(sprintf('Expected exactly one file in the archive. Found %d', $count));
+        $driverFile = null;
+        foreach ($extractedFiles as $filename) {
+            if (preg_match('/'.self::EXPECTED_DRIVER_FILE_NAME.'(.exe)?$/', basename($filename))) {
+                $driverFile = $filename;
+                break;
+            }
+        }
+
+        if ($driverFile === null) {
+            throw new UnexpectedValueException(sprintf('No driver file named "%s" found in archive', self::EXPECTED_DRIVER_FILE_NAME));
         }
 
-        $file = $this->filesystem->readlink($extractedFiles[0], true);
+        $file = $this->filesystem->readlink($driverFile, true);
```

Comment:
thought: It would probably be preferable to cover this logic in the unit tests as well. To ensure that it keeps working, and works as expected.thought: It would probably be preferable to cover this logic in the unit tests as well. To ensure that it keeps working, and works as expected.